src/StoreKeeper/WooCommerce/B2C/Exports/OrderExport.php

```diff
@@ -740,4 +637,145 @@ protected function catchKnownExceptions($throwable)
+            }
+            PaymentGateway::markPaymentAsSynced($WpObject->get_id());
+        } else { // Check if the WP order is paid and has not PaymentGateway payment.
+            // Example: the order was paid using the PayNL plugin
+            if (
+                $WpObject->is_paid() &&
+                !PaymentGateway::hasPayment($WpObject->get_id())
+            ) {
+                /**
+                 * Retrieve the current payment status of the order from the backend.
+                 * When the invoice is already payed, we should't set the payment again.
+                 */
+                $storekeeper_is_paid = (bool) $storekeeper_order['is_paid'];
+                $this->debug('Backend payment state of this order', json_encode($storekeeper_is_paid));
+
+                // Order paid in WP but not in the Backoffice.
+                if (!$storekeeper_is_paid) {
+                    // Try to mark the order as paid in the backoffice
+                    try {
+                        $PaymentModule = $this->storekeeper_api->getModule('PaymentModule');
+                        $response = $PaymentModule->listProviderMethodTypes(
+                            0,
+                            1,
+                            null,
+                            [
+                                [
+                                    'name' => 'alias__=',
+                                    'val' => 'Web',
+                                ],
+                                [
+                                    'name' => 'module_name__=',
+                                    'val' => 'PaymentModule',
+                                ],
+                            ]
+                        );
+
+                        $provider_method_type_id = null;
+                        if (empty($response['data'])) {
+                            $this->debug('No Web ProviderMethodTypes found', $storekeeper_id);
+                        } else {
+                            $provider_method_type_id = $response['data'][0]['id'];
+                        }
+
+                        $paymentGateway = wc_get_payment_gateway_by_order($WpObject);
+                        if ($paymentGateway) {
+                            $paymentGatewayTitle = $paymentGateway->get_method_title();
+                            $comment = $paymentGatewayTitle.' ('.__('Wordpress plugin').')';
+                        } else {
+                            $comment = ucwords(str_replace('pay_gateway_', '', $WpObject->get_payment_method()));
+                        }
+
+                        if (!empty($WpObject->get_meta('transactionId'))) {
+                            $comment .= ' #'.$WpObject->get_meta('transactionId');
+                        } elseif (!empty($WpObject->get_transaction_id())) {
+                            $comment .= ' #'.$WpObject->get_transaction_id();
+                        }
+
+                        $ShopModule->markOrderAsPaid(
```

Comment:
you should use the `PaymentModule/newWebPayment` for new payments, also so later you can used this id in refund. Than also no need to used the listProviderMethodTypes anymore

Reply:
Requested change implemented.